jmeter/src/main/java/org/eclipse/hono/jmeter/ui/HonoReceiverSamplerUI.java

```diff
@@ -51,22 +65,56 @@ public void modifyTestElement(final TestElement testElement) {
         super.modifyTestElement(testElement);
         HonoReceiverSampler sampler = (HonoReceiverSampler) testElement;
         sampler.setPrefetch(prefetch.getText());
+        sampler.setReconnectAttempts(reconnectAttempts.getText());
         sampler.setUseSenderTime(useSenderTime.isSelected());
+        sampler.setSenderTimeInPayload(senderTimeInPayload.isSelected());
+        sampler.setSenderTimeVariableName(senderTimeVariableName.getText());
     }
 
     @Override
     public void configure(final TestElement element) {
         super.configure(element);
         HonoReceiverSampler sampler = (HonoReceiverSampler) element;
         prefetch.setText(sampler.getPrefetch());
+        reconnectAttempts.setText(sampler.getReconnectAttempts());
         useSenderTime.setSelected(sampler.isUseSenderTime());
+        senderTimeInPayload.setSelected(sampler.isSenderTimeInPayload());
+        senderTimeVariableName.setText(sampler.getSenderTimeVariableName());
+    }
+
+    @Override
+    public void stateChanged(ChangeEvent e) {
+        if (e.getSource() == this.useSenderTime) {
+            if (this.useSenderTime.isSelected()) {
+                senderTimeInPayload.setVisible(true);
+                senderTimeVariableName.setVisible(true);
+            } else {
+                senderTimeInPayload.setVisible(false);
+                senderTimeVariableName.setVisible(false);
+            }
+        }
     }
 
     @Override
     public void clearGui() {
         super.clearGui();
+        LOGGER.debug("clearGui() invoked");
+        reconnectAttempts.setText("0");
         prefetch.setText("50");
         useSenderTime.setSelected(false);
+        senderTimeInPayload.setSelected(false);
+        senderTimeVariableName.setText("timeStamp");
     }
 
+    private JPanel createTimeStampPanel() {
+        LOGGER.debug("createTimeStampPanel() invoked");
+        JPanel timeStampPanel = new VerticalPanel();
+        timeStampPanel.setBorder(
+                BorderFactory.createTitledBorder(BorderFactory.createEtchedBorder(), "Timestamp used for sampling"));
+        useSenderTime.addChangeListener(this);
```

Comment:
I stead of using this class as the change listener for the useSenderTime UI element a lambda might be better.